crates/astria-sequencer/src/accounts/transaction.rs

```diff
@@ -3,53 +3,108 @@ use anyhow::{
     Context,
     Result,
 };
-use async_trait::async_trait;
-use borsh::{
-    BorshDeserialize,
-    BorshSerialize,
+use astria_proto::sequencer::v1::AccountsTransaction as ProtoAccountsTransaction;
+use serde::{
+    Deserialize,
+    Serialize,
 };
 use tracing::instrument;
 
-use crate::{
-    accounts::{
-        state_ext::{
-            StateReadExt,
-            StateWriteExt,
-        },
-        types::{
-            Address,
-            Balance,
-            Nonce,
-        },
+use crate::accounts::{
+    state_ext::{
+        StateReadExt,
+        StateWriteExt,
+    },
+    types::{
+        Address,
+        Balance,
+        Nonce,
     },
-    transaction::ActionHandler,
 };
 
-#[derive(BorshDeserialize, BorshSerialize, Debug, Clone, PartialEq, Eq)]
-pub(crate) struct Transaction {
-    pub(crate) to: Address,
-    pub(crate) from: Address,
-    pub(crate) amount: Balance,
-    pub(crate) nonce: Nonce,
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct Transaction {
+    to: Address,
+    amount: Balance,
+    nonce: Nonce,
+}
+
+impl Transaction {
+    pub fn new(to: Address, amount: Balance, nonce: Nonce) -> Self {
+        Self {
+            to,
+            amount,
+            nonce,
+        }
+    }
+
+    pub fn to_proto(&self) -> ProtoAccountsTransaction {
+        ProtoAccountsTransaction {
+            to: self.to.as_bytes().to_vec(),
+            amount: Some(self.amount.into()),
+            nonce: self.nonce.into(),
+        }
+    }
+
+    pub fn from_proto(proto: &ProtoAccountsTransaction) -> Result<Self> {
+        Ok(Self {
+            to: Address::try_from(proto.to.as_ref() as &[u8])?,
+            amount: proto
+                .amount
+                .as_ref()
+                .ok_or_else(|| anyhow::anyhow!("missing amount"))?
+                .into(),
+            nonce: Nonce::from(proto.nonce),
+        })
+    }
+}
+
+impl TryFrom<ProtoAccountsTransaction> for Transaction {
```

Comment:
Can we get rid of all these `TryFrom` impls in favour of just using the `from_proto` and `to_proto` methods on the types? That would make it much easier to discover these methods and how they work.

I also don't see how these methods being used in a generic context - so let's not unnecessarily introduce interfaces.